be/src/service/service_be/lake_service.cpp

```diff
@@ -114,20 +197,22 @@ void LakeServiceImpl::publish_version(::google::protobuf::RpcController* control
         return;
     }
 
-    // TODO move the execution to TaskWorkerPool
+    auto thread_pool = _env->agent_server()->get_thread_pool(TTaskType::PUBLISH_VERSION);
+    auto context = std::make_shared<PublishVersionContext>(_env, guard.release(), request, response);
+
     for (const auto& tablet_id : request->tablet_ids()) {
-        auto res = _env->lake_tablet_manager()->get_tablet(tablet_id);
-        if (!res.ok()) {
-            LOG(WARNING) << "Fail to get tablet " << tablet_id << ": " << res.status();
-            response->add_failed_tablets(tablet_id);
-            continue;
-        }
-        lake::Tablet& tablet = res.value();
-        auto st = publish(&tablet, request);
+        auto task = std::make_shared<PublishVersionTask>(tablet_id, context);
+        auto st = thread_pool->submit(std::move(task));
         if (!st.ok()) {
+            LOG(WARNING) << "Fail to submit publish version task: " << st;
+            std::lock_guard l(context->_response_mtx);
             response->add_failed_tablets(tablet_id);
         }
     }
+
+#ifdef BE_TEST
+    context->wait();
+#endif
 }
 
 void LakeServiceImpl::abort_txn(::google::protobuf::RpcController* controller,
```

Comment:
abort_txn will delete_txn_log internal, which maybe time consuming for S3 storage, so the abort_txn thread may get blocked.

Reply:
Yes you're right, I will submit another PR to fix this.